Unable to build extension

Following the steps listed in the contributing readme doc, I get an error when I run:

> 4. Build the Java modules: npm run build-plugin

The error is:
```
[INFO] Scanning for projects...
[INFO] Resolving target definition file:/Users/stephenb/personal/vscode-checkstyle/jdtls.ext/com.shengchen.checkstyle.target/target.target for environments=[linux/gtk/x86_64, win32/win32/x86_64, macosx/cocoa/x86_64], include source mode=honor, execution environment=StandardEEResolutionHints [executionEnvironment=OSGi profile 'JavaSE-17' { source level: 17, target level: 17}], remote p2 repository options=org.eclipse.tycho.p2.remote.RemoteAgent@f59da34...
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
[ERROR] Cannot resolve target definition:
[ERROR]   Software being installed: org.eclipse.jdt.ls.core 1.29.0.202310051703
[ERROR]   Missing requirement: org.eclipse.jdt.ls.core 1.29.0.202310051703 requires 'osgi.bundle; org.eclipse.lsp4j [0.21.0.0,0.22.0)' but it could not be found
[ERROR] 
[ERROR] Failed to resolve target definition file:/Users/stephenb/personal/vscode-checkstyle/jdtls.ext/com.shengchen.checkstyle.target/target.target: See log for details -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MavenExecutionException
```

Googling for the error I noticed you seem to have fixed this here: https://github.com/microsoft/vscode-java-test/pull/1507

I tried to apply that fix, but am a true beginner so didn't get very far.

--- a/jdtls.ext/com.shengchen.checkstyle.target/target.target
+++ b/jdtls.ext/com.shengchen.checkstyle.target/target.target
@@ -8,19 +8,18 @@
             <unit id="org.eclipse.equinox.sdk.feature.group" version="0.0.0"/>
             <unit id="org.eclipse.jdt.source.feature.group" version="0.0.0"/>
             <unit id="org.eclipse.sdk.feature.group" version="0.0.0"/>
-            <unit id="org.mockito.mockito-core" version="0.0.0"/>
-            <repository location="https://download.eclipse.org/eclipse/updates/4.27/R-4.27-202303020300/"/>
+            <repository location="https://download.eclipse.org/eclipse/updates/4.29/"/>
         </location>
         <location includeAllPlatforms="false" includeConfigurePhase="false" includeMode="planner" includeSource="true" type="InstallableUnit">
             <unit id="org.eclipse.xtext.xbase.lib" version="0.0.0"/>
-            <repository location="https://download.eclipse.org/releases/2022-12/"/>
+            <repository location="https://download.eclipse.org/releases/2023-09/"/>
         </location>
         <location includeAllPlatforms="false" includeConfigurePhase="false" includeMode="planner" includeSource="true" type="InstallableUnit">
             <unit id="org.eclipse.jdt.ls.core" version="0.0.0"/>
             <repository location="https://download.eclipse.org/jdtls/snapshots/repository/latest/"/>
         </location>
         <location includeAllPlatforms="false" includeConfigurePhase="false" includeMode="planner" includeSource="true" type="InstallableUnit">
-           <repository location="https://download.eclipse.org/lsp4j/updates/releases/0.20.0/"/>
+           <repository location="https://download.eclipse.org/lsp4j/updates/releases/0.21.1/"/>
            <unit id="org.eclipse.lsp4j.sdk.feature.group" version="0.0.0"/>
         </location>
     </locations>

